### PR TITLE
Add CUDA 12.6 to Linux CD docker images

### LINKED ISSUE #138440: Enable CUDA 12.6.2
### 🚀 The feature, motivation and pitch

Filing issue for enabling CUDA 12.6.2 due to an observed performance gap in PyTorch's Eager mode for plain matrix multiplications.
Confirmed that upgrading PyTorch to CTK 12.6.2 with CUBLAS 12.6.3.3 will boost the performance. 

Issue can be used to track related PRs.

cc @ptrblck @msaroufim @syed-ahmed @nWEIdia @atalman @malfet 

### Alternatives

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by tinglvv**
Reference https://github.com/pytorch/builder/pull/1003/files
Related to #138440

cc @atalman @malfet @ptrblck @nWEIdia

### CODE REVIEW

**Comment by tinglvv** [comment]
Need:
Add cuda12.6 to https://hub.docker.com/r/pytorch/libtorch-cxx11-builder/tags
Add cuda12.6 to https://hub.docker.com/r/pytorch/manylinux-builder/tags?name=cuda

File: .github/workflows/build-libtorch-images.yml [hunk 1]
```diff
@@ -44,7 +44,7 @@ jobs:
     runs-on: "${{ needs.get-label-type.outputs.label-type }}linux.9xlarge.ephemeral"
     strategy:
       matrix:
-        cuda_version: ["12.4", "12.1", "11.8"]
```
Comment (nWEIdia) [inline .github/workflows/build-libtorch-images.yml:47]: Four CUDA versions is too much for infra and unecessary. Let's remove 12.1 in this PR.

File: .github/workflows/build-manywheel-images.yml [hunk 2]
```diff
@@ -48,7 +48,7 @@ jobs:
     runs-on: "${{ needs.get-label-type.outputs.label-type }}linux.9xlarge.ephemeral"
     strategy:
       matrix:
-        cuda_version: ["12.4", "12.1", "11.8"]
```
Comment (nWEIdia) [inline .github/workflows/build-manywheel-images.yml:51]: same as above. Remove 12.1 while adding 12.6

**Review by malfet — APPROVED** [review]

**Pull request closed without merging**